### PR TITLE
Profile screen

### PR DESCRIPTION
**Pull request by DanWilkins2107**

### CODE REVIEW

File: components/Form/EditInfo.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                onChangeText={(text) => props.onChangeFunction(text)}
+            />
+            <View className="flex flex-row justify-around">
+                <TouchableOpacity
+                    className="bg-green-200 border-2 border-black rounded-3xl p-2 flex items-center justify-center w-5/12"
+                    onPress={() => {props.onSubmitFunction()}}
+                >
+                    <Text className="text-black font-bold">Submit</Text>
+                </TouchableOpacity>
+                <TouchableOpacity
```
Comment (dwulfse) [inline components/Form/EditInfo.jsx]: can we use the form button component for these? might need to allow for background/width styling in the component but could be more consistent

File: components/GeneralModal/GeneralModal.jsx [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-                    <TouchableOpacity
-                        onPress={closeModal}
-                        className="w-12 h-12 flex justify-center items-center"
-                    >
-                        <FontAwesomeIcon icon={faXmark} size={30} />
-                    </TouchableOpacity>
-                </View>
-                <View className="flex flex-1 flex-col">{children}</View>
+            <View className={`w-10/12 h-2/5 border-2 rounded-xl  bg-white`}>
+                <ScrollView className="flex flex-col" scrollEnabled={false}>
```
Comment (dwulfse) [inline components/GeneralModal/GeneralModal.jsx]: what is the benefit of a scroll view with scroll disabled?

File: components/Form/FormField.jsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import { TextInput } from "react-native";
+import { TextInput, Text } from "react-native";
```
Comment (dwulfse) [inline components/Form/FormField.jsx]: unused Text import

File: components/ProfileElements/ContactInformationModal.jsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
+import { View } from "react-native";
+import GeneralModal from "../GeneralModal/GeneralModal.jsx";
+import { firebase_auth, firebase_db } from "../../firebase.config.js";
+import { useState, useEffect } from "react";
+import EditInfo from "../Form/EditInfo.jsx";
+``;
```
Comment (dwulfse) [inline components/ProfileElements/ContactInformationModal.jsx]: dont think you need this lol

File: components/ProfileElements/ContactInformationModal.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            ) : (
+                <View className="flex flex-col items-center h-full">
+                    <EditInfo
+                        title="Contact Number"
+                        variableToEdit={contactNo}
+                        initialValue={initialContactNo}
+                        onChangeFunction={(text) => {
+                            setContactNo(text);
+                        }}
+                        onSubmitFunction={async() => {
```
Comment (dwulfse) [inline components/ProfileElements/ContactInformationModal.jsx]: this is async with no await on the set, next is async with an await and in the other document there's a non-async func with a set needs to be consistent

File: components/ProfileElements/ProfilePicture.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
+
+import { Image, Text, TouchableOpacity, View } from "react-native";
```
Comment (dwulfse) [inline components/ProfileElements/ProfilePicture.jsx]: unused touchable opacity

File: components/ProfileElements/ReportErrorModal.jsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+                <Text className="text-2xl font-extrabold text-center mb-1">Report Error</Text>
+                <View className="w-10/12 h-32 mb-6">
+                    <FormField
+                        value={errorDescription}
+                        onChangeText={setErrorDescription}
+                        extraClassName="h-32"
+                        multiline
+                    />
+                </View>
+                <TouchableOpacity
```
Comment (dwulfse) [inline components/ProfileElements/ReportErrorModal.jsx]: again, form button for this

File: components/ProfileElements/UserDetailsModal.jsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
+import { View } from "react-native";
+import GeneralModal from "../GeneralModal/GeneralModal.jsx";
+import { firebase_auth, firebase_db } from "../../firebase.config.js";
+import { useState, useEffect } from "react";
+import EditInfo from "../Form/EditInfo.jsx";
+``;
```
Comment (dwulfse) [inline components/ProfileElements/UserDetailsModal.jsx]: this weird thing again

File: components/ProfileElements/UserDetailsModal.jsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+                            set(ref(db, "users/" + auth.currentUser.uid + "/firstName"), firstName);
+                            setInitialFirstName(firstName);
+                        }}
+                    />
+                    <View className="h-8"></View>
+                    <EditInfo
+                        title="Last Name"
+                        variableToEdit={lastName}
+                        initialValue={initialLastName}
+                        onChangeFunction={(text) => setLastName(_old => text)}
```
Comment (dwulfse) [inline components/ProfileElements/UserDetailsModal.jsx]: what does _old do? I havent seen that before

File: screens/UserScreens/UserProfile.jsx [hunk 10]
```diff
@@ -1,16 +1,36 @@
...
+
 
 export default function UserProfileScreen({ navigation }) {
     const [userName, setUserName] = useState("");
-    const auth = firebase_auth;
-    const db = firebase_db;
-    
+    let [modal, setModal] = useState(null);
+    let auth = firebase_auth;
+    let db = firebase_db;
```
Comment (dwulfse) [inline screens/UserScreens/UserProfile.jsx]: can these declarations be const just cos we've used that everywhere else

File: screens/UserScreens/UserProfile.jsx [hunk 11]
```diff
@@ -1,16 +1,36 @@
...
+    const storageRef = ref(storage);
+
     useEffect(() => {
-        get(child(ref(db), `users/${auth.currentUser.uid}`)).then((snapshot) => {
+        get(child(dbRef(db), `users/${auth.currentUser.uid}`)).then((snapshot) => {
```
Comment (dwulfse) [inline screens/UserScreens/UserProfile.jsx:33]: bit confused by mix of ref (4 lines up) and dbRef?

**Review by dwulfse — CHANGES_REQUESTED** [review]
Really nice stuff - just small bits

File: components/GeneralModal/GeneralModal.jsx (hunk 2 above)
Comment (DanWilkins2107) [reply components/GeneralModal/GeneralModal.jsx]: So I can add keyboardShouldPersistTaps

File: components/Form/EditInfo.jsx (hunk 1 above)
Comment (DanWilkins2107) [reply components/Form/EditInfo.jsx]: Made SmallFormButton

File: components/ProfileElements/UserDetailsModal.jsx (hunk 9 above)
Comment (DanWilkins2107) [reply components/ProfileElements/UserDetailsModal.jsx]: _ denotes an unused variable

File: screens/UserScreens/UserProfile.jsx (hunk 11 above)
Comment (DanWilkins2107) [reply screens/UserScreens/UserProfile.jsx:33]: Spoken about

File: components/ProfileElements/UserDetailsModal.jsx (hunk 9 above)
Comment (DanWilkins2107) [reply components/ProfileElements/UserDetailsModal.jsx]: Changed anyway

File: components/ProfileElements/ContactInformationModal.jsx (hunk 5 above)
Comment (dwulfse) [reply components/ProfileElements/ContactInformationModal.jsx]: is this tested now that you got rid of the asyncs?

Comment (DanWilkins2107) [reply components/ProfileElements/ContactInformationModal.jsx]: Still need to check, but was working :)

**Review by dwulfse — COMMENTED** [review]
all changes look good I'm happy to merge as long as keyboardshouldpersisttaps and all the non-async bits work as expected

**Review by dwulfse — APPROVED** [review]

**Pull request merged**